### PR TITLE
chore: update Typescript and replace legacy build script

### PR DESCRIPTION
**Pull request by alexanderniebuhr**
⚠️ This is a stacked PR, so please keep that in mind when merging. If unsure, please ask me.

This PR updates `typescript` and all other related dependencies to their latest versions. Additionally, I removed the old build script, which was working but was rather hard to maintain. These dependencies are minimal, and `tsc` has received significant performance improvements in the past few months, so there is no strong argument for using a custom build script over `tsc`.

**Note:** The package updates do have changesets, meaning a new release might occur if the release branch is merged afterward. This is expected and should be done; however, a cross-check with `rg-workflows` is advisable.

After updating the `tsconfig.json` to a recommended preset for 2024, some oversights and bugs were discovered. I fixed them in this PR as well. There are additional improvement suggestions available, but they are out of scope for this PR.

I also removed some of the older VSCode settings. They were helpful in the past, but due to changes in recent VSCode updates, those settings are no longer ideal. If a developer wants to keep these settings, they should add them as a user profile override rather than applying them to the entire repository. In general, it is now recommended to enforce important settings, such as linting and formatting, but leave stylistic and personal settings to user profiles.

### CODE REVIEW

**Comment by alexanderniebuhr** [comment]
Who should review this?

cc @morgenstern

**Comment by morgenstern** [comment]
@sknextcode pls review this after your vacation and/or add additional reviewers.

File: package.json [hunk 1]
```diff
@@ -17,10 +17,8 @@
     "@biomejs/biome": "1.8.3",
     "@changesets/changelog-github": "0.4.7",
     "@changesets/cli": "2.25.2",
-    "@types/node": "18.7.13",
-    "esbuild": "0.15.10",
-    "tsm": "2.2.2",
-    "typescript": "4.8.4"
+    "@types/node": "22.1.0",
```
Comment (alexanderniebuhr) [inline package.json:20]:
```suggestion
    "@types/node": "18.7.13",
```

**Review by sknextcode — APPROVED** [review]
/lgtm

**Pull request merged**